packages/hoppscotch-common/src/composables/ai-experiments.ts

```diff
@@ -185,3 +185,124 @@ export const useSubmitFeedback = () => {
     isSubmitFeedbackPending,
   }
 }
+
+export const useModifyPreRequestScript = (
+  currentScript: string,
+  userPromptRef: Ref<string>,
+  generatedScriptRef: Ref<string>,
+  requestInfo: HoppRESTRequest
+) => {
+  const toast = useToast()
+  const t = useI18n()
+  const lastTraceID = ref<string | null>(null)
+  const isModifyPreRequestPending = ref(false)
+
+  const modifyPreRequestScriptForPlatform =
+    platform.experiments?.aiExperiments?.modifyPreRequestScript
+
+  const modifyPreRequestScript = async () => {
+    isModifyPreRequestPending.value = true
+
+    if (!modifyPreRequestScriptForPlatform) {
+      toast.error(t("ai_experiments.modify_prerequest_error"))
+      isModifyPreRequestPending.value = false
+      return
+    }
+
+    const result = await modifyPreRequestScriptForPlatform(
+      buildRequestInfoString(requestInfo, currentScript),
+      userPromptRef.value
+    )
+
+    if (result && E.isLeft(result)) {
+      toast.error(t("ai_experiments.modify_prerequest_error"))
+      isModifyPreRequestPending.value = false
+      return
+    }
+
+    generatedScriptRef.value = result.right.modified_script
+    lastTraceID.value = result.right.trace_id
+
+    isModifyPreRequestPending.value = false
+    return result.right
+  }
+
+  return {
+    modifyPreRequestScript,
+    isModifyPreRequestPending,
+    lastTraceID,
+  }
+}
+
+export const useModifyTestScript = (
+  currentScript: string,
+  userPromptRef: Ref<string>,
+  generatedScriptRef: Ref<string>,
+  requestInfo: HoppRESTRequest
+) => {
+  const toast = useToast()
+  const t = useI18n()
+  const lastTraceID = ref<string | null>(null)
+  const isModifyTestScriptPending = ref(false)
+
+  const modifyTestScriptForPlatform =
+    platform.experiments?.aiExperiments?.modifyTestScript
+
+  const modifyTestScript = async () => {
+    isModifyTestScriptPending.value = true
+
+    if (!modifyTestScriptForPlatform) {
+      toast.error(t("ai_experiments.modify_test_script_error"))
+      isModifyTestScriptPending.value = false
+      return
+    }
+
+    const result = await modifyTestScriptForPlatform(
+      buildRequestInfoString(requestInfo, currentScript),
+      userPromptRef.value
+    )
+
+    if (result && E.isLeft(result)) {
+      toast.error(t("ai_experiments.modify_test_script_error"))
+      isModifyTestScriptPending.value = false
+      return
+    }
+
+    generatedScriptRef.value = result.right.modified_script
+    lastTraceID.value = result.right.trace_id
+
+    isModifyTestScriptPending.value = false
+    return result.right
+  }
+
+  return {
+    modifyTestScript,
+    isModifyTestScriptPending,
+    lastTraceID,
+  }
+}
+
+const buildRequestInfoString = (
+  request: HoppRESTRequest,
+  currentScript: string
+) => {
+  return `
+  METHOD:
+  ${request.method}
+
+  URL:
+  ${request.endpoint}
+
+  BODY:
+  ${request.body.body ?? ""}
```

Comment:
When working with request.body.body, it's important to note that it isn't always a string. Direct stringification can lead to unhelpful outputs like '[Object object]', or empty objects in cases like File uploads. Additionally, we're not capturing the body's content type in the context we provide to the model.
A simple solution is to use JSON.stringify() on the body itself. This will preserve the structure and content of the request body in a string format that the LLM can better understand and process, even when dealing with complex or unexpected data types.